### PR TITLE
Add initial go templates

### PR DESCRIPTION
**Pull request by donny-dont**
Create a boilr template that creates the skeleton for a drone-plugins repository. This should hopefully simplify the task of updating a bunch of repositories whenever there are changes to the build files, docker images etc.

### CODE REVIEW

File: template/cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+)
+
+var (
+	version = "unknown"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "nuget plugin"
+	app.Usage = "nuget plugin"
```
Comment (donny-dont) [inline template/cmd/main.go]: This should be parameterized...

File: template/cmd/main.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	app.Action = run
+
+	// Create the flags for the application
+	flags := settingsFlags()
+	flags = append(flags, urfave.BuildFlags()...)
+	flags = append(flags, urfave.RepoFlags()...)
+	flags = append(flags, urfave.CommitFlags()...)
+	flags = append(flags, urfave.StageFlags()...)
+	flags = append(flags, urfave.StepFlags()...)
+	flags = append(flags, urfave.SemVerFlags()...)
```
Comment (donny-dont) [inline template/cmd/main.go]: Helper added in https://github.com/drone-plugins/drone-plugin-lib/pull/3

File: template/cmd/main.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+func run(c *cli.Context) error {
+	config := {{ Package }}.Config{
+		Build:    urfave.BuildFromContext(c),
+		Repo:     urfave.RepoFromContext(c),
+		Commit:   urfave.CommitFromContext(c),
+		Stage:    urfave.StageFromContext(c),
+		Step:     urfave.StepFromContext(c),
+		SemVer:   urfave.SemVerFromContext(c),
+		Settings: settingsFromContext(c),
+	}
```
Comment (donny-dont) [inline template/cmd/main.go]: I'm thinking we should split the Drone environment and Settings for the plugin.

See https://github.com/drone-plugins/drone-plugin-lib/pull/3

File: template/pkg/{{Package}}/plugin.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	// Config for the Plugin.
+	Config struct {
+		Build    drone_plugin.Build
+		Repo     drone_plugin.Repo
+		Commit   drone_plugin.Commit
+		Stage    drone_plugin.Stage
+		Step     drone_plugin.Step
+		SemVer   drone_plugin.SemVer
+		Settings Settings
+	}
```
Comment (donny-dont) [inline template/pkg/{{Package}}/plugin.go]: Remove after https://github.com/drone-plugins/drone-plugin-lib/pull/3

File: template/pkg/{{Package}}/plugin.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+		config Config
+	}
+)
+
+// New Plugin from the given Config.
+func New(config Config) Plugin {
+	return &pluginImpl{
+		config: config,
+	}
+}
```
Comment (donny-dont) [inline template/pkg/{{Package}}/plugin.go]: Modify after https://github.com/drone-plugins/drone-plugin-lib/pull/3

**Review by donny-dont — COMMENTED** [review]
A couple changes incoming after that PR lands but I think everyone should get the idea here with this PR.

**Comment by alikhil** [comment]
Waiting for first ready to use version :)

**Review by appleboy — APPROVED** [review]

File: template/pkg/{{Package}}/plugin.go [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+// found in the LICENSE file.
+
+package {{ Package }}
+
+import (
+	"github.com/drone-plugins/drone-plugin-lib/pkg/plugin"
+	"github.com/drone-plugins/drone-plugin-lib/pkg/urfave"
+)
+
+type pluginImpl struct {
```
Comment (tboerger) [inline template/pkg/{{Package}}/plugin.go:13]: This naming is not idiomatic Go, simply name it plugin, and let's give the plugin lib another name.

**Pull request merged**